Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -18,28 +25,114 @@ def init_locals(WorkspaceId, RegDomain):
 def Set_Marshall(WorkspaceId, Enabled, RegDomain):
     (WorkspaceId, RegDomain) = init_locals(WorkspaceId, RegDomain)
     if (Enabled):
-        i = 1
+        #call the registration script
+        try:
+            agent_id = read_oms_config_file()
+            sp = subprocess.call(
+                ["python", REGISTRATION_FILE_PATH, "--register", "-w " + WorkspaceId, "-a " + agent_id,
+                 "-c " + OMS_CERTIFICATE_PATH, "-k " + OMS_CERT_KEY_PATH, "-p " + WORKING_DIRECTORY_PATH])
+            if sp != 0:
+                LG().Log('ERROR', "Linux Hybrid Worker registration failed")
+                return [-1]
+            if not os.path.isfile(WORKER_CONF_PATH):
+                LG().Log('ERROR', "Linux Hybrid Worker registration file could not be created")
+                return [-1]
+        except Exception as exception:
+            LG().Log('ERROR', exception.message)
+
+        # Read the worker state file and try to kill linux hybrid worker process if running
+        try:
+            (pid, workspace_id) = read_worker_state()
+        except Exception as exception:
+            LG().Log('ERROR', exception.message)
+        try:
+            command = subprocess.check_output(["ps", "-p", str(pid), "-o", "comm="])
+        except subprocess.CalledProcessError:
+            pass
+        except Exception as exception:
+            LG().Log('ERROR', exception.message)
+            return [-1]
+        else:
+            # Kill the worker if it matches the description
+            if command.__contains__(WorkspaceId):
+                try:
+                    os.kill(pid, signal.SIGTERM)
+                except Exception:
+                    LG().Log('ERROR', "Could not kill Linux Hybrid Worker process")
+                    return [-1]
+        # Start the worker script
+        try:
+            sp = subprocess.call(
+                ["python", HYBRID_WORKER_START_PATH, "-c " + WORKER_CONF_PATH, "-w " + workspace_id])
+            if sp != 0:
+                LG().Log('ERROR', "Could not start Linux Hybrid Worker process")
+                return [-1]
+        except Exception as exception:
+            LG().Log('ERROR', exception)
+            return [-1]
     else:
-        #TODO: read the state file and try to see if the process is running, kill it if its runnning
+        # enabled is set to false
         try:
-            os.remove(worker_state_path)
-        except Exception:
+            (pid, workspace_id) = read_worker_state()
+            command = subprocess.check_output(["ps", "-p", str(pid), "-o", "comm="])
+        except ConfigParser.Error:
```

Comment:
In your read_worker_state please test for file existance and raise proper exception if it is missing. ConfigParser.Error is pretty broad and might swallow an exception other than the missing file.